test/test_callables.py

```diff
@@ -855,6 +855,49 @@ def test_kc_with_floor_div_in_expr(ctx_factory, inline):
     lp.auto_test_vs_ref(knl, ctx, knl)
 
 
+@pytest.mark.parametrize("inline", [True, False])
+def test_non1_step_slices(ctx_factory, inline):
+    # See https://github.com/inducer/loopy/pull/222#discussion_r645905188
+
+    ctx = ctx_factory()
+    cq = cl.CommandQueue(ctx)
+
+    callee = lp.make_function(
+            "{[i]: 0<=i<12}",
+            """
+            y[i] = i**2
+            """, name="squared_arange")
+
+    t_unit = lp.make_kernel(
+            "{[i_init, j_init]: 0<=i_init, j_init<40}",
+            """
+            X[i_init] = 42
+            X[5:40:3] = squared_arange()
+
+            Y[j_init] = 1729
+            Y[39:3:-3] = squared_arange()
+            """,
+            [lp.GlobalArg("X,Y", shape=40)],
+            seq_dependencies=True)
+
+    expected_out1 = 42*np.ones(40, dtype=np.int64)
+    expected_out1[5:40:3] = np.arange(12)**2
+
+    expected_out2 = 1729*np.ones(40, dtype=np.int64)
+    expected_out2[39:3:-3] = np.arange(12)**2
```

Comment:
Could you try this for base indices `5, 6,7` (so that all cases mod 3 are covered)? (Same for the negative stride one.)

Reply:
Yep, ced212e.